src/applications/personalization/profile-2/components/direct-deposit/DirectDepositContent.jsx

```diff
@@ -0,0 +1,311 @@
+
+import ProfileInfoTable from '../ProfileInfoTable';
+import FraudVictimAlert from './FraudVictimAlert';
+import AdditionalInformation from './DirectDepositInformation';
+
+import { prefixUtilityClasses } from '../../helpers';
+
+// TODO: move this to a common location (in platform?) for easy sharing
+/**
+ * While not included with React, this hook is described in the React Hooks
+ * documentation:
+ * https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state
+ *
+ * as well as on usehooks.com:
+ * https://usehooks.com/usePrevious/
+ *
+ * This post does a deeper dive for those who are new to hooks:
+ * https://blog.logrocket.com/how-to-get-previous-props-state-with-react-hooks/
+ */
+const usePrevious = value => {
+  const valueRef = useRef();
+  useEffect(
+    () => {
+      valueRef.current = value;
+    },
+    // only run this useEffect hook if `value` changes
+    [value],
+  );
+  // this returns _before_ the `useEffect` call completes, so this will return
+  // whatever was stored in `ref.current` _before_ this function's `useEffect`
+  // callback runs
+  return valueRef.current;
+};
+
+export const DirectDepositContent = ({
+  isAuthorized,
+  isDirectDepositSetUp,
+  directDepositAccountInfo,
+  directDepositUiState,
+  saveBankInformation,
+  toggleEditState,
+}) => {
+  const editBankInfoButton = useRef();
+  const [formData, setFormData] = useState({});
+  const [showSaveSucceededAlert, setShowSaveSucceededAlert] = useState(false);
+  const previousUiState = usePrevious(directDepositUiState);
+
+  const isEditingBankInfo = directDepositUiState.isEditing;
+
+  // when we enter and exit edit mode...
+  useEffect(
+    () => {
+      // reset the bank info form when we enter edit mode
+      if (isEditingBankInfo) {
+        setFormData({});
+      }
+      // focus the edit button when we exit edit mode
+      if (isEditingBankInfo === false) {
+        editBankInfoButton.current.focus();
+      }
+    },
+    [isEditingBankInfo],
+  );
+
+  // show the user an alert after their bank info has been updated and
+  useEffect(
+    () => {
+      if (
+        // TODO: can we add something in Redux that will flip to true after a
+        // save has succeeded and then goes back to false when we are editing
+        // bank info again?
+        previousUiState?.isSaving &&
+        !directDepositUiState.isSaving &&
+        !directDepositUiState.responseError
+      ) {
+        setShowSaveSucceededAlert(true);
+        setTimeout(() => {
+          setShowSaveSucceededAlert(false);
+        }, 6000);
+      }
+    },
+    [
+      previousUiState,
+      directDepositUiState.isSaving,
+      directDepositUiState.responseError,
+    ],
+  );
+
+  const saveBankInfo = () => {
+    // NOTE: You can mock a save error by sending undefined values in the payload
+    const payload = {
+      financialInstitutionName: 'Hidden form field',
+      financialInstitutionRoutingNumber: formData.routingNumber,
```

Comment:
Can we set a max input of 9 to the routing number and 17 max to the account? If this is out of the scope of this ticket we can do it later.

Reply:
This is all handled by the schema that's passed to the SchemaForm! https://github.com/department-of-veterans-affairs/vets-website/blob/ed1119d05ecd2ac0a24e83dcebb9f3f0fcd06589/src/applications/personalization/profile360/components/BankInfoForm.jsx#L12-L19

One thing I'm struggling with is where to test this in an integrated way. We could and should have an e2e test that ensures the form won't accept invalid input. But I'd like to do this in an "integrated unit" test by doing a full mount of the DirectDepositContent and trying to enter invalid content. That is proving to be more of a PITA than I thought :(